src/components/MessageInput/hooks/useEmojiIndex.ts

```diff
@@ -0,0 +1,22 @@
+import { useMemo } from 'react';
+import type { NimbleEmojiIndex } from 'emoji-mart';
+
+import { useChannelStateContext } from '../../../context/ChannelStateContext';
+import { useComponentContext } from '../../../context/ComponentContext';
+
+const useEmojiIndex = () => {
+  const { emojiConfig } = useChannelStateContext();
```

Comment:
technically this should have generics but since emojiConfig doesn't use them i guess it's cool. typically we've been putting them on all the time, but i guess there's not really a chance that other items will be pulled from context in this custom hook